### ISSUE TITLE
Error on start

### ISSUE THREAD

**Issue by xmha97** [Jigsaw-Code/outline-client#781]
The program cannot run properly and will encounter errors.

![image](https://user-images.githubusercontent.com/23485114/80240050-80991380-8676-11ea-9f18-1708103d0bfb.png)

![image](https://user-images.githubusercontent.com/23485114/80240071-87278b00-8676-11ea-9f94-963c3b5a6d2e.png)

**Comment by joeytwiddle** [comment]
I get this when I build my own version of Outline, but not when I download the official release.

(I have been building on Linux, running on Windows Server 2012r2 from vagrant, inside VirtualBox.)

**Comment by joeytwiddle** [comment]
@xmha97 I don't know if this will help you, but I have discovered I can use this commit: b20f67a

But building the very next commit won't connect: 9307ddc

Although in this case the error message is no more helpful than "Error" in the console!

And it's possible that later commits are fixed and working.

But the important thing is that b20f67a works for me!

(I am building with `yarn do src/electron/package_windows` not release.)

**Comment by xmha97** [comment]
Thank you for following up.
Outline has many problems, so I use Shadowsocks:
https://github.com/shadowsocks/shadowsocks-windows.

**Comment by alalamav** [comment]
Thank you for reporting this problem @xmha97.

This issue affects builds that don't define the `SENTRY_DSN` environment variable, leading the application to initialize sentry with an invalid DSN. This only seems to affect the Electron sentry SDK, as others accept an empty string as a noop. 

We plan to release two improvements to the Windows installer in the coming days. As noted by @joeytwiddle, this parituclar issue does not affect production.

**Comment by alalamav** [comment]
@joeytwiddle, could you please let us know if master (bd72dd6) builds and connects for you? Thanks!

**Comment by joeytwiddle** [comment]
Thanks for the rapid response! I just tested against your latest commit 7b61aaa. But sorry to say, I'm still getting issues.

Shortly after the first startup I see this (but not on the second run). I guess this is Sentry, and only a minor issue:
![Screenshot_20200521_190220](https://user-images.githubusercontent.com/911799/82553259-4730c680-9b96-11ea-8d25-fbb28956c0ad.png)

But when I try to connect, boom:
![Screenshot_20200521_190350](https://user-images.githubusercontent.com/911799/82553261-48fa8a00-9b96-11ea-98b8-afb98eefd380.png)

Unfortunately the log is not so helpful: 
![Screenshot_20200521_190532](https://user-images.githubusercontent.com/911799/82553264-49932080-9b96-11ea-9007-562d8a0b0c66.png)

As you can see, I'm running Windows 2012 Server in VirtualBox. But the latest production release of Outline, and my own build of old b20f67a work fine in this environment.

**Comment by alalamav** [comment]
Thank you for the detailed report, @joeytwiddle.

I can't really explain why the issue would be back in 7b61aaa. You can see that the commit didn't modify any of the electron code paths.

Regarding the connection error, there seems to be a problem with the routing service.

I can't reproduce these on my Windows 10 laptop - I'll try running VirtualBox. Are you insalling a packaged version of the client on the VM or do you build and run it directly? If it's the latter, could you please send the native electron logs?

You are hitting a [code path](https://github.com/Jigsaw-Code/outline-client/blob/master/src/electron/routing_service.ts#L183) that should only happen if you were previously connected. The error happens because we write to a socket that has been closed by the routing service. If you have some time, the following may be useful:

Check whether the routing service is running:

```
sc query outlineservice
```

Start the routing service manually (as admin):

```
net start outlineservice
```

Check the routing service logs:

```
get-eventlog -logname Application -source OutlineService -newset 20 | format-table -property timegenerated,entrytype,message -autosize
```

Thanks again for your help debugging this issue!

**Comment by joeytwiddle** [comment]
Argh. I'm sorry to report that I am also getting the HTTP error with bd72dd6, 7b61aaa and with the latest 58244e2.

Some logging shows that it is produced by `sentry.init()` inside `electron_main.ts > ElectronErrorReporter`.

![Screenshot_20200522_095908](https://user-images.githubusercontent.com/911799/82625336-70456b80-9c17-11ea-8fe6-bae09e4e1c38.png)

I addressed it in PR #800

**Comment by joeytwiddle** [comment]
Regarding the connection error.

I have mostly been building on Linux and running on Windows. I tried running the commands.

On https://app.vagrantup.com/opentable/boxes/win-2012r2-standard-amd64-nocm (Server Edition)
- Official Outline release can connect
- My build of latest Outline cannot connect
- My build of b20f67a can connect

`get-eventlog` complained about the `-newset` argument
(I should have tried without that argument but ... I didn't do that!)

On https://app.vagrantup.com/senglin/boxes/windows-10 (Windows 10 Enterprise Evaluation)
- Official Outline release cannot connect
- My build of latest Outline cannot connect
- My build of b20f67a can connect

`'get-eventlog' is not recognised as an internal or external command, operable program or batch file.`

Regarding `vc_redist` mentioned by ChrainWang. It failed to install on the Server Edition. It did install on the second Enterprise Evaluation edition (I did this before installing Outline), but it didn't improve behaviour.

Sorry that wasn't more helpful.
